ui-v2/tests/steps.js

```diff
@@ -0,0 +1,232 @@
+        return currentPage.visit();
+      })
+      .when('I visit the $name page for the "$id" $model', function(name, id, model) {
+        currentPage = pages[name];
+        return currentPage.visit({
+          [model]: id,
+        });
+      })
+      .when(
+        ['I visit the $name page for yaml\n$yaml', 'I visit the $name page for json\n$json'],
+        function(name, data) {
+          currentPage = pages[name];
+          return pages[name].visit(data);
+        }
+      )
+      .when('I click "$selector"', function(selector) {
+        return click(selector);
+      })
+      .when('I click $prop on the $component', function(prop, component) {
+        // Collection
+        var obj;
+        if (typeof currentPage[component].objectAt === 'function') {
+          obj = currentPage[component].objectAt(0);
+        } else {
+          obj = currentPage[component];
+        }
+        const func = obj[prop].bind(obj);
+        try {
+          return func();
+        } catch (e) {
+          console.error(e);
+          throw new Error(`The '${prop}' property on the '${component}' page object doesn't exist`);
+        }
+      })
+      .when('I submit', function(selector) {
+        return currentPage.submit();
+      })
+      .then('I fill in "$name" with "$value"', function(name, value) {
+        return currentPage.fillIn(name, value);
+      })
+      .then(['I fill in with yaml\n$yaml', 'I fill in with json\n$json'], function(data) {
+        return Object.keys(data).reduce(function(prev, item, i, arr) {
+          return prev.fillIn(item, data[item]);
+        }, currentPage);
+      })
+      .then(['I type with yaml\n$yaml'], function(data) {
+        const keys = Object.keys(data);
+        return keys
+          .reduce(function(prev, item, i, arr) {
+            return prev.fillIn(item, data[item]);
+          }, currentPage)
+          .then(function() {
+            return Promise.all(
+              keys.map(function(item) {
+                return triggerKeyEvent(`[name="${item}"]`, 'keyup', 83);
+              })
+            );
+          });
+      })
+      // debugging helpers
+      .then('print the current url', function(url) {
+        console.log(currentURL());
+        return Promise.resolve();
+      })
+      .then('log the "$text"', function(text) {
+        console.log(text);
+        return Promise.resolve();
+      })
+      .then('pause for $milliseconds', function(milliseconds) {
+        return new Promise(function(resolve) {
+          setTimeout(resolve, milliseconds);
+        });
+      })
+      // assertions
+      .then('a $method request is made to "$url" with the body from yaml\n$yaml', function(
+        method,
+        url,
+        data
+      ) {
+        const request = api.server.history[api.server.history.length - 2];
+        assert.equal(
+          request.method,
+          method,
+          `Expected the request method to be ${method} but was ${request.method}`
```

Comment:
This text gets displayed no matter what, which makes for a weird reading experience when the test passes:

```
Expected the request method to be PUT but was PUT
```

Reply:
Doh, cool, will change 👍

Reply:
Cool, you only really see these when they fail, unless you specifically click to show passed tests, and then open up the accordion for the test. I've tweaked the text though to make it less weird, plus entered the competition for best commit message 😄